src/js/components/question-input/question-input.js

```diff
@@ -0,0 +1,175 @@
+			content: (this.state.questionText)
+		};
+
+		this.setState({ questionArray: this.state.questionArray.concat([newQuestion])})
+		this.setState({ questionText: "" })
+		document.getElementsByName("question-textarea").value = "";
+	}
+
+	addAnswerLine() {
+		let answerType = this.state.questionType === "Code" ? "pre" : "div";
+		let newAnswer = {
+			tag: answerType,
+			class: answerType === "div" ? "text-answer" : "code-answer",
+			content: (this.state.answerText)
+		};
+
+		this.setState({ answerArray: this.state.answerArray.concat([newAnswer])})
+		this.setState({ answerText: "" })
+		document.getElementsByName("answer-textarea").value = "";
+	}
+
+	handleTextAreaKeydown(e) {
+		if (e.keyCode === 9) {
+			e.preventDefault();
+
+			let textarea = e.target;
+			let start    = textarea.selectionStart;
+			let end      = textarea.selectionEnd;
+			let value    = textarea.value;
+
+			textarea.value = value.substring(0, start) + "\t" + value.substring(end);
+
+			textarea.selectionEnd = start + 1;
+		}
+	}
+
+	render() {
+		return (
+		<div>
+			<div className="question-input__container">
+
+				 <div>
+				  	<h1>Question Info</h1>
+				  	{this.questionText}
+					<div class="question">
+						<textarea
+							name="question-textarea"
+							placeholder={ this.textareaPlaceholder }
+							onKeyDown={ this.handleTextAreaKeydown }
+							defaultValue={ this.state.questionText }
+							onChange={ this.setQuestionState.bind(this) }
+							 />
+						<br />
+					    <span>Element Type</span>
+					    <SimpleSelect
+					    	theme="default"
+					    	placeholder="Select Element Type"
+					    	options={ this.typeOptions }
+					    	defaultValue={ this.typeOptions[0] }
+					    	onValueChange={ value => this.updateQuestionType(value) } />
+						<br/>
+						<span>Element Style</span>
+						<SimpleSelect
+							theme="default"
+							placeholder="Select Element Style"
+							options={ this.viewOptions }
+							defaultValue={ this.viewOptions[0] } />
+						<br/>
+						<br/>
+					</div>
+
+					<button
+						id="question"
+						onClick={ this.addQuestionLine.bind(this) }>
+						Add Line
+					</button>
+				</div>
+
+				<div>
+					<h1>Answer Info</h1>
+					<div class="answer">
+					  	<textarea
+							name="answer-textarea"
+							placeholder={ this.textareaPlaceholder }
+							onKeyDown={ this.handleTextAreaKeydown }
+							defaultValue={ this.state.answerText }
+							onChange={ this.setAnswerState.bind(this) }
+							 />
+						<br />
+					    <span>Element Type</span>
+					    <SimpleSelect
+					    	theme="default"
+					    	placeholder="Select Element Type"
+					    	options={ this.typeOptions }
+					    	defaultValue={ this.typeOptions[0] } />
+						<br/>
+						<span>Element Style</span>
+						<SimpleSelect
+							theme="default"
+							placeholder="Select Element Style"
+							options={ this.viewOptions }
+							defaultValue={ this.viewOptions[0] } />
+						<br/>
```

Comment:
Personally, I'm under the belief of utilizing css when we need to add spacing over adding extra elements (just so the DOM is as lightweight as possible since CSS is cheap). What you do you think?